src/op_handlers/div.rs

```diff
@@ -6,14 +8,27 @@ use crate::{opcode::Opcode, state::VMState};
 pub fn div(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
     let (src0_t, src1_t) = address_operands_read(vm, opcode)?;
     let (src0, src1) = (src0_t.value, src1_t.value);
-    let (quotient, remainder) = src0.div_mod(src1);
-    if opcode.alters_vm_flags {
-        // Lt overflow is cleared
-        vm.flag_lt_of = false;
-        // Eq is set if quotient is not zero
-        vm.flag_eq = !quotient.is_zero();
-        // Gt is set if the remainder is not zero
-        vm.flag_gt = !remainder.is_zero();
+    let mut quotient = U256::zero();
+    let mut remainder = U256::zero();
+    if src1.is_zero() {
+        if opcode.alters_vm_flags {
+            // Lt overflow is set
+            vm.flag_lt_of = true;
+            // Eq is set if resultlow is 0, in this case its always 0
+            vm.flag_eq = true;
+            // Gt is set if LT_OF and EQ are cleared, they are not
+            vm.flag_gt = false;
+        }
+    } else {
+        (quotient, remainder) = src0.div_mod(src1);
```

Comment:
Maybe it's a little fussy but i think that if only these variables are used inside the condition, this declaration should be inside it, because if the `allter_flags` is not enabled, an extra operation is done that is not used.